csv.register_dialect()'s dialect arg should accept csv.Dialect subclasses

mypy 1.14 raises this error for this [code](https://github.com/pylint-dev/pylint/blob/31e7674ad44c9cbd3120e7e42497bf8af911df79/pylint/testutils/functional/lint_module_output_update.py#L19-L25), bisected to https://github.com/python/typeshed/commit/c407a444897d7c0ce3543b57f48f9d594d3fb3a0, presumably for the same reason as #12808 and #3613.

```py
error: Argument 2 to "register_dialect" has incompatible type "type[TestDialect]"; expected "type[Dialect]"  [arg-type]
```

This diff solves my use case:
```diff
diff --git a/mypy/typeshed/stdlib/_csv.pyi b/mypy/typeshed/stdlib/_csv.pyi
index 0e206a63b..21bf52303 100644
--- a/mypy/typeshed/stdlib/_csv.pyi
+++ b/mypy/typeshed/stdlib/_csv.pyi
@@ -85,7 +85,7 @@ def reader(
 ) -> _reader: ...
 def register_dialect(
     name: str,
-    dialect: type[Dialect] = ...,
+    dialect: type[Dialect | csv.Dialect] = ...,
     *,
     delimiter: str = ",",
     quotechar: str | None = '"',
```

--- a/stdlib/_csv.pyi
+++ b/stdlib/_csv.pyi
@@ -115,7 +115,7 @@ def reader(
 ) -> _reader: ...
 def register_dialect(
     name: str,
-    dialect: type[Dialect] = ...,
+    dialect: type[Dialect | csv.Dialect] = ...,
     *,
     delimiter: str = ",",
     quotechar: str | None = '"',

